### PR TITLE
MOSIP-12430

### PR DESCRIPTION
**Pull request by dhanendra06**

### CODE REVIEW

File: admin/admin-service/src/main/java/io/mosip/admin/controller/AdminController.java [hunk 1]
```diff
@@ -35,6 +30,15 @@ private ResponseWrapper<LostRidExtnDto> lostRid(@RequestBody RequestWrapper<Sear
 		auditUtil.setAuditRequestDto(EventEnum.LOST_RID_SUCCESS,null);
 		return buildLostRidResponse(lostRidResponseDto);
 	}
+
+	@GetMapping("/lostRid/details/{rid}")
```
Comment (ase-101) [inline admin/admin-service/src/main/java/io/mosip/admin/controller/AdminController.java:34]: Is this Open endpoint?

File: admin/admin-service/src/main/java/io/mosip/admin/controller/ApplicantDetailsController.java [hunk 2]
```diff
@@ -27,7 +27,7 @@ public class ApplicantDetailsController {
     @Autowired
     ApplicantDetailService applicantDetailService;
 
-    @PreAuthorize("hasRole('DIGITALCARD_ADMIN')")
+    //@PreAuthorize("hasRole('DIGITALCARD_ADMIN')")
```
Comment (ase-101) [inline admin/admin-service/src/main/java/io/mosip/admin/controller/ApplicantDetailsController.java:30]: Is this Open endpoint?

File: admin/admin-service/src/main/java/io/mosip/admin/controller/ApplicantDetailsController.java [hunk 3]
```diff
@@ -37,7 +37,7 @@ public ResponseWrapper<ApplicantDetailsDto> getApplicantDetails(@PathVariable("r
         return responseWrapper;
     }
 
-    @PreAuthorize("hasRole('DIGITALCARD_ADMIN')")
+  //  @PreAuthorize("hasRole('DIGITALCARD_ADMIN')")
```
Comment (ase-101) [inline admin/admin-service/src/main/java/io/mosip/admin/controller/ApplicantDetailsController.java:40]: why is the role check commented out?

File: admin/admin-service/src/main/java/io/mosip/admin/controller/ApplicantDetailsController.java [hunk 4]
```diff
@@ -47,7 +47,7 @@ public ResponseWrapper<ApplicantUserDetailsDto> getApplicantUserDetails() throws
         return responseWrapper;
     }
 
-    @PreAuthorize("hasRole('DIGITALCARD_ADMIN')")
+ //   @PreAuthorize("hasRole('DIGITALCARD_ADMIN')")
```
Comment (ase-101) [inline admin/admin-service/src/main/java/io/mosip/admin/controller/ApplicantDetailsController.java:50]: why is the role check commented out?

File: admin/admin-service/src/main/java/io/mosip/admin/dto/BiometricRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import lombok.EqualsAndHashCode;
+
+import java.util.List;
+
+@Data
+@EqualsAndHashCode
+public class BiometricRequestDto {
+
+    private String id;
+    private String person;
```
Comment (ase-101) [inline admin/admin-service/src/main/java/io/mosip/admin/dto/BiometricRequestDto.java:14]: what is the person property here?

File: admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java [hunk 6]
```diff
@@ -73,5 +106,93 @@ private void createLostRidRequest(SearchInfo searchInfoRequest) {
...
+		    convertRequestDto.setVersion("ISO19794_5_2011");
+			convertRequestDto.setInputBytes(isodata);
+			byte[] data = FaceDecoder.convertFaceISOToImageBytes(convertRequestDto);
+			String encodedBytes = new String(data);
+
+		return encodedBytes;
+	}
+
+	@Override
+	public LostRidDetailsDto lostRidDetails(String rid) {
```
Comment (ase-101) [inline admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java:120]: can we change the method name to getLostRidDetails

File: admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java [hunk 7]
```diff
@@ -73,5 +106,93 @@ private void createLostRidRequest(SearchInfo searchInfoRequest) {
...
+			lostRidDetailsDto.setLostRidDataMap(lostRidDataMap);
+		} catch (Exception e) {
+			throw new RequestException(LostRidErrorCode.UNABLE_TO_RETRIEVE_LOSTRID_DATA.getErrorCode(),
+					LostRidErrorCode.UNABLE_TO_RETRIEVE_LOSTRID_DATA.getErrorMessage()
+							+ e);
+		}
+		return lostRidDetailsDto;
+	}
 
+	private void getBiometric(String rid, Map<String, String> lostRidDataMap){
```
Comment (ase-101) [inline admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java:154]: change the method name to getApplicantPhoto

File: admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java [hunk 8]
```diff
@@ -73,5 +106,93 @@ private void createLostRidRequest(SearchInfo searchInfoRequest) {
...
+			byte[] data = FaceDecoder.convertFaceISOToImageBytes(convertRequestDto);
+			String encodedBytes = StringUtils.newStringUtf8(Base64.encodeBase64(data, false));
+			String imageData = "data:image/png;base64," + encodedBytes;
+			if(response!=null && responseObj.get("response")==null) {
+				throw new RequestException(ApplicantDetailErrorCode.RID_NOT_FOUND.getErrorCode(),
+						ApplicantDetailErrorCode.RID_NOT_FOUND.getErrorMessage());
+			}
+			lostRidDataMap.put("applicantPhoto",imageData);
+		} catch (Exception e) {
+			throw new RuntimeException(e);
```
Comment (ase-101) [inline admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java:178]: log the error also

File: admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java [hunk 9]
```diff
@@ -73,5 +106,93 @@ private void createLostRidRequest(SearchInfo searchInfoRequest) {
...
+		}
+
+	}
```
Comment (ase-101) [inline admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java:181]: Provide atleast 1 new line gap between methods

File: admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java [hunk 10]
```diff
@@ -73,5 +106,93 @@ private void createLostRidRequest(SearchInfo searchInfoRequest) {
...
+	private void buildBiometricRequestDto(BiometricRequestDto biometricRequestDto, String rid) {
+		List<String> modalities=new ArrayList<>();
+		biometricRequestDto.setSource(SOURCE);
+		biometricRequestDto.setId(rid);
+		biometricRequestDto.setProcess(PROCESS);
+		biometricRequestDto.setPerson(INDIVIDUAL_BIOMETRICS);
+		modalities.add("Face");
+		biometricRequestDto.setModalities(modalities);
+	}
```
Comment (ase-101) [inline admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java:190]: Provide atleast 1 new line gap between methods

**Review by ase-101 — CHANGES_REQUESTED** [review]

File: admin/admin-service/src/main/java/io/mosip/admin/constant/LostRidErrorCode.java [hunk 11]
```diff
@@ -2,7 +2,8 @@
 
 public enum LostRidErrorCode {
 
-	UNABLE_TO_RETRIEVE_LOSTRID("ADMN-LRID-001", "Unable ro find the lost rid..");
+	UNABLE_TO_RETRIEVE_LOSTRID("ADMN-LRID-001", "Unable to find the lost rid.."),
+	UNABLE_TO_RETRIEVE_LOSTRID_DATA("ADMN-LRID-001", "Unable to find the lost rid data");
```
[the remark this answers is not in the document]
Comment (dhanendra06) [reply admin/admin-service/src/main/java/io/mosip/admin/constant/LostRidErrorCode.java]: taken care

File: admin/admin-service/src/main/java/io/mosip/admin/controller/AdminController.java (hunk 1 above)
Comment (dhanendra06) [reply admin/admin-service/src/main/java/io/mosip/admin/controller/AdminController.java:37]: taken care

File: admin/admin-service/src/main/java/io/mosip/admin/controller/ApplicantDetailsController.java (hunk 3 above)
Comment (dhanendra06) [reply admin/admin-service/src/main/java/io/mosip/admin/controller/ApplicantDetailsController.java:41]: taken care

File: admin/admin-service/src/main/java/io/mosip/admin/controller/ApplicantDetailsController.java (hunk 4 above)
Comment (dhanendra06) [reply admin/admin-service/src/main/java/io/mosip/admin/controller/ApplicantDetailsController.java:52]: taken care

File: admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java (hunk 7 above)
Comment (dhanendra06) [reply admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java]: taken care

File: admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java (hunk 8 above)
Comment (dhanendra06) [reply admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java]: taken care

File: admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java (hunk 9 above)
Comment (dhanendra06) [reply admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java:183]: taken care

File: admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java (hunk 10 above)
Comment (dhanendra06) [reply admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java:194]: taken care

File: admin/admin-service/src/main/java/io/mosip/admin/dto/BiometricRequestDto.java (hunk 5 above)
Comment (ase-101) [reply admin/admin-service/src/main/java/io/mosip/admin/dto/BiometricRequestDto.java:14]: you already have a separate property List<String> modalities, what is set in individualBiometrics aka person property?

File: admin/admin-service/src/main/java/io/mosip/admin/dto/FieldDtos.java [hunk 12]
```diff
@@ -0,0 +1,17 @@
+package io.mosip.admin.dto;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+import java.util.List;
+
+@Data
+@EqualsAndHashCode
+public class FieldDtos {
```
Comment (ase-101) [inline admin/admin-service/src/main/java/io/mosip/admin/dto/FieldDtos.java:10]: why is it named as FieldDtos? why is it plural and also it has fields property in it .. its confusing @dhanendra06

Comment (ase-101) [reply admin/admin-service/src/main/java/io/mosip/admin/dto/FieldDtos.java:10]: Based on the review AdminServiceImpl, I think it is good to rename FieldDtos to SearchFieldsRequestDto.

File: admin/admin-service/src/main/java/io/mosip/admin/dto/FieldResponseDto.java
Comment (ase-101) [inline admin/admin-service/src/main/java/io/mosip/admin/dto/FieldResponseDto.java:1]: Also Rename this to SearchFieldsResponseDto

File: admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java [hunk 13]
```diff
@@ -27,6 +37,29 @@ public class AdminServiceImpl implements AdminService {
...
 
+	@Value("${mosip.admin.lostrid.details.fields:fullName,dateOfBirth}")
+	private String[] fields;
+
+	@Value("${mosip.admin.lostrid.details.name.field:fullName}")
+	private String nameField;
+
+	private static final String PROCESS = "NEW";
+
+	private static final String INDIVIDUAL_BIOMETRICS = "individualBiometrics";
```
Comment (ase-101) [inline admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java:48]: can we take this from configuration

File: admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java [hunk 14]
```diff
@@ -73,5 +106,99 @@ private void createLostRidRequest(SearchInfo searchInfoRequest) {
...
+		LostRidDetailsDto lostRidDetailsDto=new LostRidDetailsDto();
+		Map<String,String> lostRidDataMap=new HashMap<>();
+		FieldDtos fieldDtos=new FieldDtos();
+		RequestWrapper<FieldDtos> fieldDtosRequestWrapper=new RequestWrapper<>();
+		ConvertRequestDto convertRequestDto = new ConvertRequestDto();
+		try {
+			FieldResponseDto fieldResponseDto=new FieldResponseDto();
+			buildSearchFieldsRequestDto(fieldDtos,rid);
+			fieldDtosRequestWrapper.setRequest(fieldDtos);
+			ResponseWrapper<FieldDtos> fieldDtosResponseWrapper = restClient.postApi(ApiName.PACKET_MANAGER_SEARCHFIELDS, MediaType.APPLICATION_JSON,
```
Comment (ase-101) [inline admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java:130]: We are assigning the response to ResponseWrapper<FieldDtos>  but later we are trying to deserialize the response to FieldResponseDto. 
Why are we defining the FieldResponseDto again and again.

File: admin/admin-service/src/main/java/io/mosip/admin/dto/BiometricRequestDto.java (hunk 5 above)
Comment (dhanendra06) [reply admin/admin-service/src/main/java/io/mosip/admin/dto/BiometricRequestDto.java:14]: its request @ase-101 we can't change the attribute

File: admin/admin-service/src/main/java/io/mosip/admin/dto/FieldDtos.java (hunk 12 above)
Comment (dhanendra06) [reply admin/admin-service/src/main/java/io/mosip/admin/dto/FieldDtos.java]: taken care

File: admin/admin-service/src/main/java/io/mosip/admin/dto/FieldResponseDto.java
Comment (dhanendra06) [reply admin/admin-service/src/main/java/io/mosip/admin/dto/FieldResponseDto.java:1]: taken care

File: admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java (hunk 13 above)
Comment (dhanendra06) [reply admin/admin-service/src/main/java/io/mosip/admin/service/impl/AdminServiceImpl.java:51]: taken care

File: admin/admin-service/src/main/java/io/mosip/admin/dto/BiometricRequestDto.java (hunk 5 above)
Comment (ase-101) [reply admin/admin-service/src/main/java/io/mosip/admin/dto/BiometricRequestDto.java:14]: ok

**Review by ase-101 — APPROVED** [review]

**Pull request merged**